### ISSUE TITLE
Failed to launch games on Steam Deck

### ISSUE THREAD

**Issue by raygan** [redromnon/HeroicBashLauncher#40]
I can launch games from the Heroic launcher, but trying to launch them via these scripts results in an error that says "Failed to launch games. Consider posting the log as an issue."

So that's what I'm doing! It looks like it's not finding the "launch" command for some reason? This is a new out of the box Steam Deck and I'm not sure what the issue could be. Here's my log:

```
Gtk-Message: 14:05:39.085: Failed to load module "canberra-gtk-module"
3.32.0
sh: line 3: launch: command not found
---CANNOT CONNECT TO NETWORK. RUNNING IN OFFLINE MODE---
sh: line 3: launch: command not found
Gtk-Message: 14:05:39.134: Failed to load module "canberra-gtk-module"

** (zenity:10): WARNING **: 14:05:39.285: AT-SPI: Could not obtain desktop path or name

a
** (zenity:10): WARNING **: 14:05:39.300: AT-SPI: Could not obtain desktop path or name


** (zenity:10): WARNING **: 14:05:39.300: atk-bridge: get_device_events_reply: unknown signature

** (zenity:10): WARNING **: 14:05:39.300: atk-bridge: get_device_events_reply: unknown signature

** (zenity:10): WARNING **: 14:05:39.300: atk-bridge: GetRegisteredEvents returned message with unknown signature

** (zenity:10): WARNING **: 14:05:39.315: AT-SPI: Could not obtain desktop path or name


** (zenity:10): WARNING **: 14:05:39.315: atk-bridge: get_device_events_reply: unknown signature

** (zenity:10): WARNING **: 14:05:39.315: atk-bridge: get_device_events_reply: unknown signature

** (zenity:10): WARNING **: 14:05:39.315: atk-bridge: GetRegisteredEvents returned message with unknown signature
Using Bash Launcher 2.4.2

Checking if Zenity is installed:
Launch command:
STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH='/home/deck/Games/Heroic/Prefixes/DEMONS-TILT' launch b8b37813c5634dbdac61d88ff411135c --no-wine --wrapper "'/home/deck/.steam/steam/steamapps/common/Proton - Experimental/proton' run" 
```

**Comment by moogletimes** [comment]
I'm having the same issue on top of some weird Locale issue i don't understand.

Here are my logs...


```
(process:4): Gtk-WARNING **: 01:19:20.895: Locale not supported by C library.
	Using the fallback 'C' locale.
Gtk-Message: 01:19:20.898: Failed to load module "canberra-gtk-module"
3.32.0
sh: line 3: launch: command not found
---CANNOT CONNECT TO NETWORK. RUNNING IN OFFLINE MODE---
sh: line 3: launch: command not found

(process:10): Gtk-WARNING **: 01:19:20.937: Locale not supported by C library.
	Using the fallback 'C' locale.
Gtk-Message: 01:19:20.940: Failed to load module "canberra-gtk-module"
Using Bash Launcher 2.4.2

Checking if Zenity is installed:
Launch command:
STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH='/home/deck/Games/Heroic/Prefixes/FINAL-FANTASY-VII-REMAKE-INTERGRADE' launch b25ba0f04dc9434f8a04a062719e7aa7 --override-exe '/home/deck/Games/FFVIIRemakeIntergrade/ff7remake.exe' --no-wine --wrapper "'/home/deck/.steam/steam/steamapps/common/Proton 7.0/proton' run" 
```

**Comment by nhughes118** [comment]
Same issue when trying to launch KH3
[KINGDOMHEARTSIIIReMind_Heroic.log](https://github.com/redromnon/HeroicBashLauncher/files/8499069/KINGDOMHEARTSIIIReMind_Heroic.log)

<pre>
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Gtk-Message: 21:39:15.197: Failed to load module "canberra-gtk-module"
3.32.0
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
sh: line 3: launch: command not found
pid 7 != 5, skipping destruction (fork without exec?)
---CANNOT CONNECT TO NETWORK. RUNNING IN OFFLINE MODE---
sh: line 3: launch: command not found
pid 9 != 5, skipping destruction (fork without exec?)
pid 8 != 5, skipping destruction (fork without exec?)
pid 6 != 5, skipping destruction (fork without exec?)
ERROR: ld.so: object '/home/deck/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Gtk-Message: 21:39:15.292: Failed to load module "canberra-gtk-module"

** (zenity:10): WARNING **: 21:39:15.445: AT-SPI: Could not obtain desktop path or name


** (zenity:10): WARNING **: 21:39:15.456: AT-SPI: Could not obtain desktop path or name


** (zenity:10): WARNING **: 21:39:15.456: atk-bridge: get_device_events_reply: unknown signature

** (zenity:10): WARNING **: 21:39:15.456: atk-bridge: get_device_events_reply: unknown signature

** (zenity:10): WARNING **: 21:39:15.456: atk-bridge: GetRegisteredEvents returned message with unknown signature

** (zenity:10): WARNING **: 21:39:15.468: AT-SPI: Could not obtain desktop path or name


** (zenity:10): WARNING **: 21:39:15.468: atk-bridge: get_device_events_reply: unknown signature

** (zenity:10): WARNING **: 21:39:15.468: atk-bridge: get_device_events_reply: unknown signature

** (zenity:10): WARNING **: 21:39:15.468: atk-bridge: GetRegisteredEvents returned message with unknown signature
Using Bash Launcher 2.4.2

Checking if Zenity is installed:
Launch command:
STEAM_COMPAT_CLIENT_INSTALL_PATH=/home/deck/.var/app/com.heroicgameslauncher.hgl/.steam/steam STEAM_COMPAT_DATA_PATH='/home/deck/Games/Heroic/Prefixes/KINGDOM-HEARTS-III--Re-Mind' launch fd711544a06543e0ab1b0808de334120 --no-wine --wrapper "'/home/deck/.steam/steam/steamapps/common/Proton 5.13/proton' run" 

</pre>

**Comment by redromnon** [comment]
Yeah, it's a mistake on my end. Will release it the fix soon.

**Comment by redromnon** [comment]
It should be fixed in [v2.4.3](https://github.com/redromnon/HeroicBashLauncher/releases/tag/v2.4.3)

**Comment by raygan** [comment]
Thanks, this resolved the issue for me.
